PR: https://github.com/odoo/odoo/pull/

From: 3ae9dde258303b8212ae47654edf00e5879e8bab
From: Prakash Prajapati
Date: 2018-04-10 07:00:17

Structural Changes: 4
Total Changes: 4

[IMP] mrp: new filters on products

There was a filter 'Bom Components', that didn't filter anything. So we
make it work to show the products/product templates that are used as
components in BOM.

We've also added a filter 'Manufactured Products' that show the
products/product templates that have at least one BOM

TASK-54350

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_bom.py
+++ b/addons/mrp/models/mrp_bom.py
@@ -184,6 +184,7 @@ class MrpBomLine(models.Model):
 
     product_id = fields.Many2one(
         'product.product', 'Product', required=True)
+    product_tmpl_id = fields.Many2one('product.template', 'Product Template', related='product_id.product_tmpl_id')
     product_qty = fields.Float(
         'Product Quantity', default=1.0,
         digits=dp.get_precision('Product Unit of Measure'), required=True)

--- a/addons/mrp/models/product.py
+++ b/addons/mrp/models/product.py
@@ -7,6 +7,7 @@ from odoo import api, fields, models
 class ProductTemplate(models.Model):
     _inherit = "product.template"
 
+    bom_line_ids = fields.One2many('mrp.bom.line', 'product_tmpl_id', 'BoM Components')
     bom_ids = fields.One2many('mrp.bom', 'product_tmpl_id', 'Bill of Materials')
     bom_count = fields.Integer('# Bill of Material', compute='_compute_bom_count')
     used_in_bom_count = fields.Integer('# of BoM Where is Used', compute='_compute_used_in_bom_count')
@@ -49,6 +50,8 @@ class ProductTemplate(models.Model):
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
+    variant_bom_ids = fields.One2many('mrp.bom', 'product_id', 'BOM Product Variants')
+    bom_line_ids = fields.One2many('mrp.bom.line', 'product_id', 'BoM Components')
     bom_count = fields.Integer('# Bill of Material', compute='_compute_bom_count')
     used_in_bom_count = fields.Integer('# BoM Where Used', compute='_compute_used_in_bom_count')
     mo_count = fields.Integer('# Manufacturing Orders', compute='_compute_mo_count')
